WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -64,100 +92,199 @@ struct WooShippingAddPackageView: View {
+                                    Text(Localization.keyboardDoneButton)
+                                        .bold()
+                                }
+                            }
+                            .renderedIf(focusedField != nil)
                         }
                     }
+                    Toggle(isOn: $showSaveTemplate) {
+                        Text(Localization.saveNewPackageTemplate)
+                            .font(.subheadline)
+                    }
+                    .tint(Color(.withColorStudio(.wooCommercePurple, shade: .shade60)))
+                    if showSaveTemplate {
+                        TextField("Enter a unique package name", text: $packageTemplateName)
+                            .font(.body)
+                            .focused($packageTemplateNameFieldFocused)
+                            .padding()
+                            .roundedBorder(cornerRadius: 8,
+                                           lineColor: packageTemplateNameFieldFocused ? Color(UIColor.wooCommercePurple(.shade60)) : Color(.separator),
+                                           lineWidth: packageTemplateNameFieldFocused ? 2 : 1)
+                        Button(Localization.savePackageTemplate) {
+                            savePackageAsTemplateButtonTapped()
+                        }
+                        .disabled(!validateCustomPackageInputFields())
+                        .buttonStyle(SecondaryButtonStyle())
+                        .padding(.bottom)
+                        .id(Constants.saveTemplateButtonID) // Set the id for the button so we can scroll to it
+                    }
                 }
-            } label: {
-                HStack {
-                    // text
-                    Text(packageType.name)
-                        .bodyStyle()
-                    // arrows
-                    Spacer()
-                    Image(systemName: "chevron.up.chevron.down")
+                .padding(.horizontal)
+                .onChange(of: showSaveTemplate) { newValue in
+                    packageTemplateNameFieldFocused = newValue
                 }
-                .padding()
-            }
-            .roundedBorder(cornerRadius: 8, lineColor: Color(.separator), lineWidth: 1)
-            AdaptiveStack(spacing: 8) {
-                ForEach(WooShippingAddPackageDimensionView.DimensionType.allCases, id: \.self) { dimensionType in
-                    WooShippingAddPackageDimensionView(dimensionType: dimensionType, fieldValue: Binding(get: {
-                        return self.fieldValues[dimensionType] ?? ""
-                    }, set: { value in
-                        self.fieldValues[dimensionType] = value
-                    }))
+                .onChange(of: packageTemplateNameFieldFocused) { focused in
+                    if focused {
+                        DispatchQueue.main.asyncAfter(deadline: .now() + Constants.scrollToDelay, execute: {
```

Comment:
What's the reasoning behind the delay for this scroll action? It feels fine in testing, I'm just curious if it's needed for some reason or if it's just for the feel.

Reply:
When text field for name gets focused (packageTemplateNameFieldFocused changed) the SwiftUI automatically scrolls so that text field is visible, but the save button is not, and if I do not do the scrolling to button with a small delay this is basically called in paralel as scrolling to text field so save button is overlayed a bit with keyboard.
This is how it goes without a delay (half of the button is visible):

https://github.com/user-attachments/assets/a459a87e-a60c-45e9-a3ff-da6a8e70e81e

Reply:
Ah yeah, that makes sense. Thanks for the explanation! Maybe it's worth a small note in the code about that so that later on we remember why it's needed?